### PR TITLE
[BACKLOG-40444] Browse Perspective: Improve folder tree model depth variable and related API call

### PR DESCRIPTION
**Pull request by varuntangirala**
This PR is part of https://github.com/pentaho/pentaho-scheduler-plugin/pull/176

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | junit:junit:4.12 | junit:junit 4.12 | [4.13.1] | CVE-2020-15250 |

</div>

### 🔬 Research Details
**Description:**
In JUnit4 from version 4.7 and before 4.13.1, the test rule TemporaryFolder contains a local information disclosure vulnerability. On Unix like systems, the system's temporary directory is shared between all users on that system. Because of this, when files and directories are written into this directory they are, by default, readable by other users on that same system. This vulnerability does not allow other users to overwrite the contents of these directories or files. This is purely an information disclosure vulnerability. This vulnerability impacts you if the JUnit tests write sensitive information, like API keys or passwords, into the temporary folder, and the JUnit tests execute in an environment where the OS has other untrusted users. Because certain JDK file system APIs were only added in JDK 1.7, this this fix is dependent upon the version of the JDK you are using. For Java 1.7 and higher users: this vulnerability is fixed in 4.13.1. For Java 1.6 and lower users: no patch is available, you must use the workaround below. If you are unable to patch, or are stuck running on Java 1.6, specifying the `java.io.tmpdir` system environment variable to a directory that is exclusively owned by the executing user will fix this vulnerability. For more information, including an example of vulnerable code, see the referenced GitHub Security Advisory.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | junit:junit:4.12 | junit:junit 4.12 | [4.13.1] | CVE-2020-15250 |

</div>

### 🔬 Research Details
**Description:**
In JUnit4 from version 4.7 and before 4.13.1, the test rule TemporaryFolder contains a local information disclosure vulnerability. On Unix like systems, the system's temporary directory is shared between all users on that system. Because of this, when files and directories are written into this directory they are, by default, readable by other users on that same system. This vulnerability does not allow other users to overwrite the contents of these directories or files. This is purely an information disclosure vulnerability. This vulnerability impacts you if the JUnit tests write sensitive information, like API keys or passwords, into the temporary folder, and the JUnit tests execute in an environment where the OS has other untrusted users. Because certain JDK file system APIs were only added in JDK 1.7, this this fix is dependent upon the version of the JDK you are using. For Java 1.7 and higher users: this vulnerability is fixed in 4.13.1. For Java 1.6 and lower users: no patch is available, you must use the workaround below. If you are unable to patch, or are stuck running on Java 1.6, specifying the `java.io.tmpdir` system environment variable to a directory that is exclusively owned by the executing user will fix this vulnerability. For more information, including an example of vulnerable code, see the referenced GitHub Security Advisory.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 12m 29s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | junit:junit:4.12 | junit:junit 4.12 | [4.13.1] | CVE-2020-15250 |

</div>

### 🔬 Research Details
**Description:**
In JUnit4 from version 4.7 and before 4.13.1, the test rule TemporaryFolder contains a local information disclosure vulnerability. On Unix like systems, the system's temporary directory is shared between all users on that system. Because of this, when files and directories are written into this directory they are, by default, readable by other users on that same system. This vulnerability does not allow other users to overwrite the contents of these directories or files. This is purely an information disclosure vulnerability. This vulnerability impacts you if the JUnit tests write sensitive information, like API keys or passwords, into the temporary folder, and the JUnit tests execute in an environment where the OS has other untrusted users. Because certain JDK file system APIs were only added in JDK 1.7, this this fix is dependent upon the version of the JDK you are using. For Java 1.7 and higher users: this vulnerability is fixed in 4.13.1. For Java 1.6 and lower users: no patch is available, you must use the workaround below. If you are unable to patch, or are stuck running on Java 1.6, specifying the `java.io.tmpdir` system environment variable to a directory that is exclusively owned by the executing user will fix this vulnerability. For more information, including an example of vulnerable code, see the referenced GitHub Security Advisory.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>  Medium | junit:junit:4.12 | junit:junit 4.12 | [4.13.1] | CVE-2020-15250 |

</div>

### 🔬 Research Details
**Description:**
In JUnit4 from version 4.7 and before 4.13.1, the test rule TemporaryFolder contains a local information disclosure vulnerability. On Unix like systems, the system's temporary directory is shared between all users on that system. Because of this, when files and directories are written into this directory they are, by default, readable by other users on that same system. This vulnerability does not allow other users to overwrite the contents of these directories or files. This is purely an information disclosure vulnerability. This vulnerability impacts you if the JUnit tests write sensitive information, like API keys or passwords, into the temporary folder, and the JUnit tests execute in an environment where the OS has other untrusted users. Because certain JDK file system APIs were only added in JDK 1.7, this this fix is dependent upon the version of the JDK you are using. For Java 1.7 and higher users: this vulnerability is fixed in 4.13.1. For Java 1.6 and lower users: no patch is available, you must use the workaround below. If you are unable to patch, or are stuck running on Java 1.6, specifying the `java.io.tmpdir` system environment variable to a directory that is exclusively owned by the executing user will fix this vulnerability. For more information, including an example of vulnerable code, see the referenced GitHub Security Advisory.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 13m
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 59s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by NJtwentyone — APPROVED** [review]

File: user-console/src/main/resources-filtered/org/pentaho/mantle/public/browser/index.jsp [hunk 1]
```diff
@@ -269,7 +269,20 @@
         else if (paramJson.eventSubType == "RefreshBrowsePerspectiveEvent") {
           //Clear the Browse Perspective cache
           window.parent.mantle_isBrowseRepoDirty = true;
-          FileBrowser.update(window.parent.HOME_FOLDER, paramJson.booleanParam); // refresh folder list
+          var path;
+          var folderClicked = FileBrowser.fileBrowserModel.getFolderClicked();
```
Comment (dcleao) [inline user-console/src/main/resources-filtered/org/pentaho/mantle/public/browser/index.jsp:273]: I believe this logic is already being done in `FileBrowser.update`. Please check, possibly fix a root bug, if there is one.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js [hunk 2]
```diff
@@ -295,20 +297,22 @@ define([
...
+
+      var _clickedFolder = {
+        obj: foldersTreeModel ? foldersTreeModel.get("clickedFolder") : fileBrowserClickedFolder,
+        time: (new Date()).getTime()
+      };
+
+      var _clickedFile = {
+        obj: fileListModel ? fileListModel.get("clickedFile") : fileBrowserClickedFile,
+        time: (new Date()).getTime()
+      };
```
Comment (dcleao) [inline user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js:314]: Would be better to have something like:

```js
var _clickedFolderObj = (foldersTreeModel && foldersTreeModel.get("clickedFolder")) || 
         (myself.get("clickedFolder") && myself.get("clickedFolder").obj);

var _clickedFolder = _clickedFolderObj ? {obj: _clickedFolderObj, time: (new Date()).getTime()} : null;
```

Then, you shouldn't need to be testing for `clickedFolder && clickedFolder.obj` elsewhere.
I see you also test for `clickedFolder.obj.attr("path")` elsewhere. If that's really needed, then it seems that should also be included here.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl user-console
```
---
:ok_hand: All tests passed!
 
Tests run: **12**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5609/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by rmansoor — APPROVED** [review]

**Pull request merged**